Leave/leave_interface.py

```diff
@@ -119,3 +119,15 @@ def AddRetroactiveLeaveToDB(member, message_id, start_date, end_date, leave_type
             annual_leave_balance -= 1
         leave_db.InsertLeave(member.id, message_id, leave_type, day, reason, "", leave_status, is_emergency, is_unpaid)
     return ("Retroactive leave was inserted successfully")
+
+def IsMemberWorking(member_id, date):
+    work_days = utils.GetWorkDays(date, date)
+    if len(work_days) <= 0:
+        return (False, "Holiday/Weekend")
+
+    applied_leaves = list(filter(lambda leave: leave['date'] == date and leave['leave_status'] != "Approved", leave_db.GetLeavesByMemberID(member_id)))
+    if len(applied_leaves) > 0:
+        reason = applied_leaves[0]["reason"]
+        return (False, reason)
```

Comment:
Helper function for this probably.

Reply:
Done

Reply:
Meant a helper function for `IsMemberOnLeave`. The way you did it is much less flexible and doesn't exactly achieve anything with regards to readability or reusability.

Reply:
can you check now ?